### PR TITLE
Add JACK support to example applications

### PR DESCRIPTION
**Pull request by rncbc**
Add new LINK_BUILD_JACK option to build example applications with JACK support on Linux (http://jackaudio.org).

### CODE REVIEW

**Comment by abique** [comment]
Does it bring transport master/slave to jack?

**Comment by rncbc** [comment]
nope. afaict. Link is not a transport service. but rather a shared metronome.
that said it might relate to  jack timebase though.

**Comment by abique** [comment]
So jack transport is already superior to link?

**Comment by rncbc** [comment]
> So jack transport is already superior to link?

i guess not. they are somewhat orthogonal in function and purpose. Link is about to syncing apps to a common shared tempo on beat (quanta) musical time resolution. jack transport is about state (play/stop) and transport positioning in absolute sample (frame) real time resolution. Link'ed apps do play, when they play, on the same tempo and beat and phase. on Jack transport all clients start/stop at the same time and on the same and absolute frame, hopefully :)

File: examples/linkaudio/AudioPlatform_Jack.cpp [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+int AudioPlatform::audioCallback(jack_nframes_t nframes)
+{
+  using namespace std::chrono;
+  AudioEngine& engine = mEngine;
+
+  const auto hostTime = mHostTimeFilter.sampleTimeToHostTime(mSampleTime);
+
+  mSampleTime += nframes;
+
+  const auto bufferBeginAtOutput = hostTime + engine.mOutputLatency;
```
Comment (fgo-ableton) [inline examples/linkaudio/AudioPlatform_Jack.cpp]: `mOutputLatency` is never set. AFAIK there is no way to query this directly from JACK as it actually depends on the host driver. So it is probably ok to just derive it from the buffer size.

Comment (rncbc) [inline examples/linkaudio/AudioPlatform_Jack.cpp]: this last commit https://github.com/Ableton/link/pull/5/commits/1afff04 fixes that.

thanks.

**Comment by fgo-ableton** [comment]
@rncbc Sorry. Accidentally pushed to your remote. I just squashed you branch to clean it up. I'd merge this way if this is ok for you.

**Pull request merged**